### PR TITLE
Feature: new wallet flow

### PR DESCRIPTION
**Pull request by adrian-tiberius**
Implements new wallet flow.

### CODE REVIEW

**Comment by rasom** [comment]
1.  `request-transaction` label

screen shot 2016-11-01 at 12 00 23

2. Maybe it is better to use `?` when it is an answer to the request and `!` when it is just a command

screen shot 2016-11-01 at 12 02 15

3. 

screen shot 2016-11-01 at 12 04 49

4. Can not scan QR

qr

**Comment by rasom** [comment]
These two bugs are not related to this PR, but they appear because of it :) 
I will create separate issues

1. 

screen shot 2016-11-01 at 12 03 41


2. Request icon is overlapped by the next message 
screen shot 2016-11-01 at 12 38 38

**Comment by jarradh** [comment]
regarding 1, good question. I don't like the idea of truncating the name but we will have to. 

and 2 (1?) I guess request message should have a minimum height that eliminates the overlap

**Comment by jarradh** [comment]
both the request message and the send eth message will have better designs

**Comment by jarradh** [comment]
for example, at the bottom of this https://zpl.io/Z1cQdF4

**Comment by jarradh** [comment]
regarding earlier comment:
- yes !send should be !send.. not sure why this has changed, it's not using the !send command ?
- yes the parameters should be prefilled
- although we can think of a better name than `request-transaction` we should establish a limit., I also notice that we're missing the 'pretty' command titles and instead still filling with the !command name

**Comment by adrian-tiberius** [comment]
- changed request-transaction to request (this doesn't solve the long command names problem which I think needs to be discussed in a different issue about dapp api specs)
- response prefill is related to general handling of requests/responses and I think it would be cleaner to tackle it in a different issue/pr
- fixed qr scan

File: src/status_im/accounts/styles.cljs [hunk 1]
```diff
@@ -113,3 +113,56 @@
...
+  {:align-items :center
+   :padding-vertical 40})
+
+(def footer
+  {:position :absolute
+   :left     0
+   :right    0
+   :bottom   0
+   :height 166
+   :background-color "#2f3031"})
```
Comment (alwx) [inline src/status_im/accounts/styles.cljs]: code formatting for hashmaps is a bit strange, but it is not a big deal, of course

File: src/status_im/chat/handlers/requests.cljs [hunk 2]
```diff
@@ -15,6 +16,7 @@
                  :type       (:command content)
                  :added      (js/Date.)}
         request' (update request :type keyword)]
+    (log/debug "Adding request: " request')
```
Comment (alwx) [inline src/status_im/chat/handlers/requests.cljs]: do we need all these log/debug calls everywhere?

File: src/status_im/data_store/test_data/contacts.cljs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    :whisper-identity "test_identity_9"
+    :name             "Johny Cash"}])
+
+(defn add-test-contacts
+  []
+  (log/debug "Contacts added already ?"(data-store/exists? "test_identity_1"))
+  (when-not (data-store/exists? "test_identity_1")
+    (doseq [contact contacts]
+      (log/debug "Adding contact " contact)
+      (data-store/save contact false))))
```
Comment (alwx) [inline src/status_im/data_store/test_data/contacts.cljs]: Do we need the test data to be merged to develop branch?

Comment (jarradh) [inline src/status_im/data_store/test_data/contacts.cljs]: no

File: src/status_im/chat/handlers/requests.cljs (hunk 2 above)
Comment (adrian-tiberius) [inline src/status_im/chat/handlers/requests.cljs]: will be removed before squashing if everything is ok.

**Comment by rasom** [comment]
I see this error after scanning QR

screenshot_2016-11-04-17-14-10

**Comment by rasom** [comment]
Send request

screen shot 2016-11-04 at 16 53 58

but receive response

screenshot_2016-11-04-16-53-55

**Pull request merged**